nexus/reconfigurator/execution/src/datasets.rs

```diff
@@ -133,11 +259,33 @@ mod tests {
     type ControlPlaneTestContext =
         nexus_test_utils::ControlPlaneTestContext<omicron_nexus::Server>;
 
+    fn get_all_datasets_from_zones(
+        blueprint: &Blueprint,
+    ) -> Vec<BlueprintDatasetConfig> {
+        blueprint
+            .all_omicron_zones(BlueprintZoneFilter::All)
+            .filter_map(|(_, zone)| {
+                if let Some(dataset) = zone.zone_type.durable_dataset() {
```

Comment:
Tiny nit - I think this could be

```suggestion
                let dataset = zone.zone_type.durable_dataset()?;
```

to save a level of indentation

Reply:
Updated